src/Http/Routing/src/EndpointRoutingMiddleware.cs

```diff
@@ -107,7 +109,16 @@ private Task SetRoutingAndContinue(HttpContext httpContext)
                 Log.MatchSuccess(_logger, endpoint);
             }
 
-            return _next(httpContext);
+            try
+            {
+                await _next(httpContext);
+            }
+            finally
+            {
+                // We unset the endpoint after calling through to the next middleware. This enables any future calls into
+                // endpoint routing don't no-op from there already being an endpoint set.
+                httpContext.SetEndpoint(endpoint: null);
```

Comment:
Do we need to clear route values as well?

Reply:
I'm tempted to leave this as is until we see a need. It looks like all of the EndpointRouting bits override whatever is there when it attempts to re-match. If you feel different let me know and I can give it a shot

Reply:
Clear both.

Reply:
Done.